app/core/rate_limiting/agent_wrapper.py

```
@@ -0,0 +1,392 @@
+                            limits=rate_check.metadata.dict(),
+                            model=self.model
+                        )
+                    )
+                raise RateLimitExceededException(
+                    message=f"Rate limit exceeded for {self.model}",
+                    retry_after=rate_check.retry_after,
+                    limits=rate_check.metadata.dict(),
+                    model=self.model
+                )
+            
+            self.logger.info(f"✅ Rate limit check passed, proceeding with stream")
```

Flake8 (7.2.0)

[error] f-string is missing placeholders

(F541)